In [7]:
###Question1
import pandas as pd
dataframe = pd.read_csv("Travel Pony Facebook.csv")
# print(dataframe)
Amount_spend=dataframe["Amount Spent (USD)"]
Impression=dataframe["Impressions"]
CostPerImpression=Amount_spend/Impression
dataframe.insert(loc=0, column="CostPerImpression", value=CostPerImpression)
# print(list(dataframe))
import arrow
DAY=[]
for i in range(len(dataframe)):
    dt=arrow.get(dataframe['Start Date'].iloc[i], 'M/D/YY')
    Dayoftheweek = dt.isoweekday()
    DAY.append(Dayoftheweek)

dataframe.insert(loc=0,value=DAY,column='Day of the week')
Final=dataframe.groupby(['Day of the week'])['CostPerImpression'].mean()
Maximum=max(dataframe.groupby(['Day of the week'])['CostPerImpression'].mean())
Minimum=min(dataframe.groupby(['Day of the week'])['CostPerImpression'].mean())
print(Maximum)
print(Minimum)
print(Final)

0.004096890719487211
0.0026286969333697923
Day of the week
1    0.002910
2    0.003040
3    0.002981
4    0.003398
5    0.004097
6    0.002629
7    0.003688
Name: CostPerImpression, dtype: float64


Highest cost:Friday         
Lowest cost:Saturday


In [8]:
Question2a=dataframe["Amount Spent (USD)"].corr(dataframe['Reach'])
Question2b=dataframe["Amount Spent (USD)"].corr(dataframe['Frequency'])
Question2c=dataframe["Amount Spent (USD)"].corr(dataframe['Unique Clicks'])
Question2d=dataframe["Amount Spent (USD)"].corr(dataframe['Page Likes'])
print(Question2a)
print(Question2b)
print(Question2c)
print(Question2d)

0.7031238065113846
0.13020086992866337
0.8829931774784137
0.7576119292180449


#Answer:Unique Clicks=0.883, so it has strongest correlation. 
#Unick clicks means that when people interested in the advertisement, they will click it. And it means that these people most likely buy the product.

In [10]:
import statsmodels.api as SSMD
Question3a=dataframe["Unique Clicks"]
Question3b=dataframe[["Reach","Frequency"]]
QuestionM= SSMD.OLS(Question3a, Question3b).fit()
QuestionP= QuestionM.predict(Question3b) 
print(QuestionM.summary())

/Users/yifanli/anaconda3/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


                            OLS Regression Results                            
Dep. Variable:          Unique Clicks   R-squared:                       0.557
Model:                            OLS   Adj. R-squared:                  0.556
Method:                 Least Squares   F-statistic:                     2325.
Date:                Thu, 18 Oct 2018   Prob (F-statistic):               0.00
Time:                        16:06:10   Log-Likelihood:                -15973.
No. Observations:                3705   AIC:                         3.195e+04
Df Residuals:                    3703   BIC:                         3.196e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Reach          0.0019   3.12e-05     62.490      0.0

#Answer for question3: Frequency has higher coefficient which is 3.029 than Reach, thus Frequency most strongly predicts unique clicks. 
#when you see the same advertisement multiple times, you have greater possibility to buy this product. 
